pkg/scheduler/framework/plugins/noderesources/least_allocated.go

```diff
@@ -63,25 +64,46 @@ func (la *LeastAllocated) ScoreExtensions() framework.ScoreExtensions {
 }
 
 // NewLeastAllocated initializes a new plugin and returns it.
-func NewLeastAllocated(_ runtime.Object, h framework.FrameworkHandle) (framework.Plugin, error) {
+func NewLeastAllocated(laArgs runtime.Object, h framework.FrameworkHandle) (framework.Plugin, error) {
+	args := kubeschedulerconfig.NodeResourcesLeastAllocatedArgs{}
+
+	if err := framework.DecodeInto(laArgs, &args); err != nil {
+		return nil, err
+	}
+
+	resToWeightMap := make(resourceToWeightMap)
+	if len(args.Resources) == 0 {
+		// If no resources specified, used the default set.
+		resToWeightMap = defaultRequestedRatioResources
+	}
+	for _, resource := range args.Resources {
+		resToWeightMap[v1.ResourceName(resource.Name)] = resource.Weight
+		if resource.Weight == 0 {
```

Comment:
This should be documented in the API doc comments.

Reply:
what if the list only contains (say) CPU? then memory would be excluded. Is that the intent?

Reply:
if the list only contains CPU, looks like to me it's okay to assume only CPU is concerned, memory should not be accountable for the final score, this is the only way we can inspect user's intention.  btw, this also align with plugin of "RequestedToCapacityRatio".

the alternative is we can set the default weight for those defined in "defaultRequestedRatioResources" but not defined in args. I am not sure which one is better.

Reply:
the most important things regarding a change such as this one are:
- Make sure people not using the new fields don't get any behavior change
- The API is properly documented to explain how things are used.

Reply:
> the most important things regarding a change such as this one are:
> 
> * Make sure people not using the new fields don't get any behavior change

yes, this is new args for the plugin, if it's not configured, it will fall back to the default config.
> * The API is properly documented to explain how things are used.

Added some comments here, https://github.com/kubernetes/kubernetes/pull/90544/files#diff-154828dfe0f8fb7ffbb5e779ce6360ebR67-R68 , I guess this is not enough,  can you point me which doc I need to update then?

Reply:
See other comment. However, here you are putting 1 for resources that set 0. Is the assumption that if you want to exclude resources you would just exclude them from the list?

If we are going that path, set the value in defaults.go